src/components/App.js

```diff
@@ -51,6 +65,17 @@ class App extends React.Component {
       );
     }
   }
-};
+  else if(this.state.isRulesVisible){ 
+    return (<div className = "rules">
```

Comment:
<div className = "rules"> can be within Rules component itself?
I understand that you have written it this way so as to accommodate the new button, but it doesn't make sense to have two separate buttons as well. Just have one button, outside both these if conditions and change the contents of the button based on isBoardVisible or isRulesVisible

Reply:
but if the button is outside of the if conditions then how will the button will render. it renders the board page and never renders the global button because isBoardVisible is set to true by default.

Reply:
anything outside the if conditions would always be rendered in app.js right? only the contents have to be changed appropriately. Also, isBoardVisible's value changes on clicking the button right? It's not always true, it just starts out like that.

Reply:
anything outside the if conditions will always be rendered if previous if conditions didn't get executed, In code after entering the username and creating room 

```
if(isBoardVisible){   //This will always got executed because it is by default true"** 
   // Renders the board
}
if(isRulesVisible)  //By default false when the button in above if condition clicked state changes and isRulesVisible exec
{   
   // Renders Rules Page
}

```
now tell me where I put the button and how I render the button.

Reply:
Ah, okay, sorry, I didn't realise that either way we had to return a component in either of those cases, the two button method is fine, however, move the if condition to the else after checking user, room.
```
     if (username === "") {
      return (<Username />);
    } else if (room === "") {
      return (<Room socket={this.socket} />)
    } else {
      if(this.state.isBoardVisible){
```